src/per/adc.cpp

```diff
@@ -353,22 +291,22 @@ void AdcHandle::Init(AdcChannelConfig* cfg,
     sConfig.Offset       = 0;
     for(uint8_t i = 0; i < adc.channels; i++)
     {
-        const auto& cfg = adc.pin_cfg[i];
+        auto& cfg = adc.pin_cfg[i];
```

Comment:
why remove the const here?

Reply:
We need to do `cfg.pin.Init()` for all of those GPIO in cfg, which means we can't use const.